src/Components/Patient/PatientHome.tsx

```diff
@@ -1315,16 +1316,55 @@ export const PatientHome = (props: any) => {
               </div>
               <div
                 className="w-full"
-                onClick={() => setOpenAssignVolunteerDialog(true)}
+                onClick={() => {
+                  if (
+                    !(
+                      !patientData.is_active ||
+                      !(patientData?.last_consultation?.facility === facilityId)
+                    )
```

Comment:
```suggestion
                    !(
                      !patientData.is_active ||
                      !(patientData?.last_consultation?.facility === facilityId)
                    )
```
It seems that this snippet of code is being repeated multiple times, can it be made into a function?

Reply:
Done!